sdk/python/lib/pulumi/runtime/mocks.py

```diff
@@ -21,22 +21,30 @@
 from typing import Dict, List, NamedTuple, Optional, Tuple, TYPE_CHECKING
 
 from google.protobuf import empty_pb2
-from . import rpc
+from . import rpc, rpc_manager
 from .settings import Settings, configure, get_stack, get_project, get_root_resource
 from .sync_await import _ensure_event_loop, _sync_await
 from ..runtime.proto import engine_pb2, provider_pb2, resource_pb2
-from ..runtime.stack import Stack, run_pulumi_func
+from ..runtime.stack import Stack, run_pulumi_func, wait_for_rpcs
 
 if TYPE_CHECKING:
     from ..resource import Resource
 
 
 def test(fn):
+    """
+    Decorates a test function to make sure that a returned Future
+    or Output is awaited as part of the test.
+    """
 
     @functools.wraps(fn)
     def wrapper(*args, **kwargs):
         from .. import Output  # pylint: disable=import-outside-toplevel
-        _sync_await(run_pulumi_func(lambda: _sync_await(Output.from_input(fn(*args, **kwargs)).future())))
+
+        try:
+            _sync_await(run_pulumi_func(lambda: _sync_await(Output.from_input(fn(*args, **kwargs)).future())))
+        finally:
```

Comment:
If this is too much blast radius, I can make it local to the test that need it. It does however seem reasonable to increase test isolation.